components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/fetch/GeckoViewFetchClient.kt

```diff
@@ -57,6 +57,7 @@ class GeckoViewFetchClient(
             var fetchFlags = 0
             if (request.cookiePolicy == Request.CookiePolicy.OMIT) {
                 fetchFlags += GeckoWebExecutor.FETCH_FLAGS_ANONYMOUS
+                fetchFlags += GeckoWebExecutor.FETCH_FLAGS_PRIVATE
```

Comment:
@pocmo not sure if this is the right approach, I could also add `PRIVATE` to `CookiePolicy` but I'm not sure how it should be handle on OKHTTP client:

    enum class CookiePolicy {
        /**
         * Include cookies when sending the request.
         */
        INCLUDE,

        /**
         * Do not send cookies with the request.
         */
        OMIT,
       
        /**
         * Associates the request a private browsing session.
         */
        PRIVATE
    }

Reply:
Yeah, let's add "private". For okhttp: We could eventually implement our own "in-memory" `CookieHandler` or `CookieJar`. But for now let's just file it and map it to using no cookies in okhttp?

Reply:
Ok, sounds good 👍🏽

Reply:
Hm I would handle this with a separate flag passed to `Request`, because it covers more than just cookies. If we add it to `CookiePolicy` only then it's also unclear what the difference is compared to `OMIT`. As a consume that would confuse me...

Reply:
That's a good point. I didn't look closely at what it does... if it's more than cookies then yeah, let's have a separate flag. 👍

Reply:
I updated the pr adding `private` as part of the request :)